infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/Orchestrator.java

```diff
@@ -75,6 +81,30 @@ void handleQuery(CorfuPayloadMsg<OrchestratorRequest> msg, ChannelHandlerContext
                 .payloadMsg(new OrchestratorResponse(resp)));
     }
 
+    void removeNode(CorfuPayloadMsg<OrchestratorRequest> msg, ChannelHandlerContext ctx, IServerRouter r) {
+        RemoveNodeRequest req = (RemoveNodeRequest) msg.getPayload().getRequest();
+
+        if (findUUID(req.getEndpoint()) != null) {
+            // An add node workflow is already executing for this endpoint, return
+            // existing workflow id.
+            OrchestratorResponse resp = new OrchestratorResponse(
+                    new CreateWorkflowResponse(findUUID(req.getEndpoint())));
+            r.sendResponse(ctx, msg, CorfuMsgType.ORCHESTRATOR_RESPONSE
+                    .payloadMsg(resp));
+            return;
+        } else {
+            // Create a new workflow for this endpoint and return a new workflow id
+            Workflow workflow = getWorkflow(msg.getPayload());
+            activeWorkflows.put(workflow.getId(), req.getEndpoint());
+            OrchestratorResponse resp = new OrchestratorResponse(new CreateWorkflowResponse(workflow.getId()));
+            r.sendResponse(ctx, msg, CorfuMsgType.ORCHESTRATOR_RESPONSE
+                    .payloadMsg(resp));
+            CompletableFuture.runAsync(() -> {
```

Comment:
I think you should first start the asyncWorkflow before sending response. Just for sanity.

Reply:
Done.